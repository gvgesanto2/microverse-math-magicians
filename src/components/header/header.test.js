import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Header from './header.component';

let component;

describe('Header component', () => {
  beforeEach(() => {
    component = render(<Header />, { wrapper: BrowserRouter });
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should navigate to the homepage when clicking the header\'s logo', () => {
    const LogoLink = screen.getByRole('link', { name: /math magician/i });

    userEvent.click(LogoLink);

    expect(window.location.pathname).toBe('/');
  });

  it('should navigate to the homepage when clicking the home navigation link', () => {
    const HomeNavLink = screen.getByRole('link', { name: 'home' });

    userEvent.click(HomeNavLink);

    expect(window.location.pathname).toBe('/');
  });

  it('should navigate to the calculator page when clicking the calculator navigation link', () => {
    const CalculatorNavLink = screen.getByRole('link', { name: 'calculator' });

    userEvent.click(CalculatorNavLink);

    expect(window.location.pathname).toBe('/calculator');
  });

  it('should navigate to the quote page when clicking the quote navigation link', () => {
    const QuoteNavLink = screen.getByRole('link', { name: 'quote' });

    userEvent.click(QuoteNavLink);

    expect(window.location.pathname).toBe('/quote');
  });
});
