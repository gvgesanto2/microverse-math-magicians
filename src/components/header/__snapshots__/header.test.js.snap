// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header__content"
        >
          <a
            data-testid="logo-link"
            href="/"
          >
            <span
              class="header__logo"
            >
              math magicians
            </span>
          </a>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li>
                <a
                  class="header__nav-link"
                  href="/"
                >
                  home
                </a>
              </li>
              <li>
                <a
                  class="header__nav-link"
                  href="/calculator"
                >
                  calculator
                </a>
              </li>
              <li>
                <a
                  class="header__nav-link"
                  href="/quote"
                >
                  quote
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="header__hamburger-container"
          >
            <div
              class="hamburger-menu "
            >
              <button
                aria-label="Open hamburger menu"
                class="hamburger-btn "
                type="button"
              >
                <span
                  class="hamburger-btn__icon"
                >
                   
                </span>
              </button>
              <div
                class="hamburger-menu__background"
              >
                 
              </div>
              <nav
                class="hamburger-menu__nav"
              >
                <ul
                  class="hamburger-menu__nav-list"
                >
                  <li>
                    <a
                      class="hamburger-menu__link"
                      href="/"
                    >
                      <span>
                        01
                      </span>
                      home
                    </a>
                  </li>
                  <li>
                    <a
                      class="hamburger-menu__link"
                      href="/calculator"
                    >
                      <span>
                        02
                      </span>
                      calculator
                    </a>
                  </li>
                  <li>
                    <a
                      class="hamburger-menu__link"
                      href="/quote"
                    >
                      <span>
                        03
                      </span>
                      quote
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </header>
      <main
        class="main"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header__content"
      >
        <a
          data-testid="logo-link"
          href="/"
        >
          <span
            class="header__logo"
          >
            math magicians
          </span>
        </a>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li>
              <a
                class="header__nav-link"
                href="/"
              >
                home
              </a>
            </li>
            <li>
              <a
                class="header__nav-link"
                href="/calculator"
              >
                calculator
              </a>
            </li>
            <li>
              <a
                class="header__nav-link"
                href="/quote"
              >
                quote
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="header__hamburger-container"
        >
          <div
            class="hamburger-menu "
          >
            <button
              aria-label="Open hamburger menu"
              class="hamburger-btn "
              type="button"
            >
              <span
                class="hamburger-btn__icon"
              >
                 
              </span>
            </button>
            <div
              class="hamburger-menu__background"
            >
               
            </div>
            <nav
              class="hamburger-menu__nav"
            >
              <ul
                class="hamburger-menu__nav-list"
              >
                <li>
                  <a
                    class="hamburger-menu__link"
                    href="/"
                  >
                    <span>
                      01
                    </span>
                    home
                  </a>
                </li>
                <li>
                  <a
                    class="hamburger-menu__link"
                    href="/calculator"
                  >
                    <span>
                      02
                    </span>
                    calculator
                  </a>
                </li>
                <li>
                  <a
                    class="hamburger-menu__link"
                    href="/quote"
                  >
                    <span>
                      03
                    </span>
                    quote
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </header>
    <main
      class="main"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
